.github/workflows/main.yml

```diff
@@ -8,16 +8,22 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.config.os }}
     timeout-minutes: 30
+    defaults:
+      run:
+        shell: ${{ matrix.config.shell }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        config:
+          - { os: macOS-latest, shell: bash }
```

Comment:
Oddly enough `bash` is the default for macOS (though Apple already changed the default to `zsh`).
`zsh` is not even installed with the MacOS [image](https://github.com/actions/virtual-environments/blob/a91dbe89493b1661224afeff1d898e8e9a87e3e2/images/macos/macos-10.15-Readme.md).